tests/test_cuda.py

```diff
@@ -23,6 +26,32 @@ def test_export_engine():
     YOLO(f)(SOURCE, device=0)
 
 
+@pytest.mark.slow
+@pytest.mark.skipif(not CUDA_IS_AVAILABLE, reason="CUDA is not available")
+@pytest.mark.parametrize(
+    "task, dynamic, int8, half, batch",
+    [  # generate all combinations but exclude those where both int8 and half are True
+        (task, dynamic, int8, half, batch)
+        for task, dynamic, int8, half, batch in product(TASKS, [True, False], [True, False], [True, False], [1, 2])
+        if not (int8 and half)  # exclude cases where both int8 and half are True
```

Comment:
This is tough, because I'd love to run this full matrix daily, but our current hardware availability is just not there for support yet (memory maybe, but utilization no). I think it might make sense to put in place a dedicated server fo CI for this, which would also allow us to expand to additional runners. 

I think for now let's cut this back to the bare minimum, i.e. just 1 export.

<img width="990" alt="Screenshot 2024-05-08 at 15 44 08" src="https://github.com/ultralytics/ultralytics/assets/26833433/f226ced3-2683-4eed-93f8-55968cf2164f">